prowler/providers/gcp/services/compute/compute_service.py

```diff
@@ -78,60 +80,58 @@ def __get_projects__(self):
                     f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
                 )
 
-    def __get_instances__(self):
+    def __get_instances__(self, zone):
         for project_id in self.project_ids:
             try:
-                for zone in self.zones:
-                    request = self.client.instances().list(
-                        project=project_id, zone=zone
-                    )
-                    while request is not None:
-                        response = request.execute()
-
-                        for instance in response.get("items", []):
-                            public_ip = False
-                            for interface in instance["networkInterfaces"]:
-                                for config in interface.get("accessConfigs", []):
-                                    if "natIP" in config:
-                                        public_ip = True
-                            self.instances.append(
-                                Instance(
-                                    name=instance["name"],
-                                    id=instance["id"],
-                                    zone=zone,
-                                    public_ip=public_ip,
-                                    metadata=instance["metadata"],
-                                    shielded_enabled_vtpm=instance[
-                                        "shieldedInstanceConfig"
-                                    ]["enableVtpm"],
-                                    shielded_enabled_integrity_monitoring=instance[
-                                        "shieldedInstanceConfig"
-                                    ]["enableIntegrityMonitoring"],
-                                    confidential_computing=instance.get(
-                                        "confidentialInstanceConfig", {}
-                                    ).get("enableConfidentialCompute", False),
-                                    service_accounts=instance.get(
-                                        "serviceAccounts", []
-                                    ),
-                                    ip_forward=instance.get("canIpForward", False),
-                                    disks_encryption=[
-                                        (
-                                            disk["deviceName"],
-                                            True
-                                            if disk.get("diskEncryptionKey", {}).get(
-                                                "sha256"
-                                            )
-                                            else False,
+                request = self.client.instances().list(project=project_id, zone=zone)
+                http = google_auth_httplib2.AuthorizedHttp(
```

Comment:
Can you move this to the `GCPService` class, you can create a function like `__get_AuthorizedHttp_client__`.

Reply:
Done